docs/spark-connect-overview.md

```diff
@@ -0,0 +1,108 @@
+
+# How to use Spark Connect
+
+Starting with Spark 3.4, Spark Connect is available and supports PySpark applications. When creating a Spark session, you can specify that you want to use Spark Connect and there are a few ways to do that as outlined below.
+
+If you do not use one of the mechanisms outlined below, your Spark session will work just like before, without leveraging Spark Connect, and your application code will run on the Spark driver node.
+
+## Set SPARK_REMOTE environment variable
+
+If you set the SPARK_REMOTE environment variable on the client machine where your Spark client application is running and create a new Spark Session as illustrated below, the session will be a Spark Connect session. With this approach, there is no code change needed to start using Spark Connect.
+
+Set SPARK_REMOTE environment variable:
+
+{% highlight bash %}
+    export SPARK_REMOTE="sc://localhost/"
+{% endhighlight %}
+
+Start the PySpark CLI, for example:
+
+{% highlight bash %}
+    ./bin/pyspark
+{% endhighlight %}
+
+And notice that the PySpark CLI is now connected to Spark using Spark Connect as indicated in the welcome message: “Client connected to the Spark Connect server at...”.
+
+And if you write your own Python program, create a Spark Session as shown in this example:
+
+{% highlight python %}
+    from pyspark.sql import SparkSession
+    spark = SparkSession.builder.getOrCreate()
+{% endhighlight %}
+
+Which will create a Spark Connect session by reading the SPARK_REMOTE environment variable we set above.
+
+## Specify Spark Connect when creating Spark session
+
+You can also specify that you want to use Spark Connect when you create a Spark session explicitly.
+
+For example, when launching the PySpark CLI, simply include the remote parameter as illustrated here:
+
+{% highlight bash %}
+    ./bin/pyspark --remote "sc://localhost"
+{% endhighlight %}
+
+And again you will notice that the PySpark welcome message tells you that you are connected to Spark using Spark Connect.
+
+Or, in your code, include the remote function with a reference to your Spark server when you create a Spark session, as in this example:
+
+{% highlight python %}
+    from pyspark.sql import SparkSession
+    spark = SparkSession.builder.remote("sc://localhost/").getOrCreate()
+{% endhighlight %}
+
+# Client application authentication
+
+While Spark Connect does not have built-in authentication, it is designed to work seamlessly with your existing authentication infrastructure. Its gRPC HTTP/2 interface allows for the use of authenticating proxies, which makes it possible to secure Spark Connect without having to implement authentication logic in Spark directly.
+
+# What is included in Spark 3.4
+
+In Spark 3.4, Spark Connect provides DataFrame API coverage for PySpark and 
+Spark Connect clients for other languages are planned for the future.
```

Comment:
I believe it is important to inform users in advance that while Spark Connect supports many of the key features of PySpark, there are also some features that are not supported.

For example, we can write something like this:

In Spark 3.4, Spark Connect supports most of the key APIs of PySpark, including DataFrame, Functions, and Column. However, some APIs such as SparkContext and RDD are not supported yet. The support for specific PySpark APIs in Spark Connect is indicated in all PySpark API references with phrases like “Support Spark Connect”. Therefore, it is recommended to check in advance whether the APIs you are using are supported by Spark Connect before migrating existing code. Support for Spark Connect clients in other languages is planned for the future.

the below is including the link as well just for your convenience (when you built the docs, the link would work):

```md
In Spark 3.4, Spark Connect supports most of the key APIs of PySpark, including [DataFrame](api/python/reference/pyspark.sql/dataframe.html), [Functions](api/python/reference/pyspark.sql/functions.html), and [Column](api/python/reference/pyspark.sql/column.html). However, some APIs such as [SparkContext](api/python/reference/api/pyspark.SparkContext.html#pyspark.SparkContext) and [RDD](api/python/reference/api/pyspark.RDD.html#pyspark.RDD) are not supported yet. The support for specific PySpark APIs in Spark Connect is indicated in all PySpark API references with phrases like "Support Spark Connect". Therefore, it is recommended to check in advance whether the APIs you are using are supported by Spark Connect before migrating existing code. Support for Spark Connect clients in other languages is planned for the future.
```

Reply:
I added this content, thanks